### DISCUSSION TITLE
Open with...

### DISCUSSION THREAD

**Discussion by oong819** [Sketchware-Pro/Sketchware-Pro#769] [Q&A] [answered]
i made my app appear on Open with dialog, but how to get the opened file path?
![Screenshot_20230218-170345](https://user-images.githubusercontent.com/76146430/219854416-23cb5a7f-bced-43c8-9aec-09b388fb06d1.png)
![Screenshot_20230218-165759](https://user-images.githubusercontent.com/76146430/219854420-d4e412d5-55a8-4e0a-a0bd-904ea0ce870f.png)

**Comment by j-px** [comment]
[See this Android Developers guide](https://developer.android.com/training/secure-file-sharing#HandleIntent).

**Comment by oong819** [comment]
i tried this (get MIME type):
![Screenshot_20230219-001828](https://user-images.githubusercontent.com/76146430/219879375-0274c65c-a203-4fa6-8f09-933df914942d.png)
![Screenshot_20230219-001832](https://user-images.githubusercontent.com/76146430/219879382-afc74b60-ceb7-41d4-8798-9ac47ce5ade9.png)

any code snippet or example? i tried some code from that doc

also recently my app have 32 classes.dex but i dont include any library, how to remove it?

**Reply by PranavPurwar** [reply]
> i tried this (get MIME type): ![Screenshot_20230219-001828](https://user-images.githubusercontent.com/76146430/219879375-0274c65c-a203-4fa6-8f09-933df914942d.png) ![Screenshot_20230219-001832](https://user-images.githubusercontent.com/76146430/219879382-afc74b60-ceb7-41d4-8798-9ac47ce5ade9.png)
> 
> any code snippet or example? i tried some code from that doc
> 
> also recently my app have 32 classes.dex but i dont include any library, how to remove it?

i think you have enabled appcompat or something

**Reply by oong819** [reply]
i once enable it but i disabled it, also found a lot of third party library too

**Reply by oong819** [reply]
maybe its because of bumptech glide, i removed the setimageurl and its gone

**Comment by DevVigilante** [comment, marked as answer]
Use :
```java
Uri uri = getIntent().getData();
```
instead of 
```java
Uri uri = fileLoader.getData();
```

**Reply by oong819** [reply]
work now! thx
